src/plugins/crypto/gpg.c

```diff
@@ -784,25 +784,37 @@ int CRYPTO_PLUGIN_FUNCTION (gpgCall) (KeySet * conf, Key * errorKey, Key * msgKe
 		if (msgKey)
 		{
 			close (STDIN_FILENO);
-			dup (pipe_stdin[0]);
+			if (dup (pipe_stdin[0]) < 0)
+			{
+				ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "failed to redirect stdin.");
+				exit (42);
+			}
 		}
 		close (pipe_stdin[0]);
 
 		// redirect stdout to pipe
 		close (STDOUT_FILENO);
-		dup (pipe_stdout[1]);
+		if (dup (pipe_stdout[1]) < 0)
+		{
+			ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "failed to redirect the stdout.");
+			exit (42);
+		}
 		close (pipe_stdout[1]);
 
 		// redirect stderr to pipe
 		close (STDERR_FILENO);
-		dup (pipe_stderr[1]);
+		if (dup (pipe_stderr[1]) < 0)
+		{
+			ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "failed to redirect stderr.");
+			exit (42);
+		}
 		close (pipe_stderr[1]);
 
 		// finally call the gpg executable
 		if (execv (argv[0], argv) < 0)
 		{
 			ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "failed to start the gpg binary: %s", argv[0]);
-			return -1;
+			exit (42);
```

Comment:
If 42 is used everywhere, you cannot distinguish in the parent process which error happened.